src/data/zmod/basic.lean

```diff
@@ -219,36 +219,36 @@ by { cases n; refl }
 
 end
 
-lemma nat_cast_surjective [fact (0 < n)] :
-  function.surjective (coe : ℕ → zmod n) :=
+lemma nat_cast_right_inverse [fact (0 < n)] :
```

Comment:
This name sounds like `nat_cast` is the right inverse...

Reply:
I don't know what the naming convention is for these... I can currently find these lemmas that seem to match what I think you would suggest:
* `right_inverse_pth_root_frobenius`
* `prod.swap_right_inverse`
* `equiv.right_inverse_symm`
Along with the lemmas that I added recently, which seem to match what I've done here:
* `free_algebra.algebra_map_left_inverse`
* `tensor_algebra.ι_left_inverse`

Reply:
[Zulip](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/Naming.20of.20.60right_inverse.60.20lemmas/near/223148651)

Reply:
Can we say that this is blocked on #5829, and then come back to it when that is resolved either way?

Reply:
I'd rather not - I'm not sure how useful #5892 actually is, whereas with things like #6167 now merged, having readily-available inverse proofs is immediately useful.

Reply:
I still think that the names used in this PR are confusing. If that issue can be solved, this PR is ready for merging.